tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -4310,6 +4310,216 @@ TEST_F(OpConverterTest, ConvertConv3D) {
                 ElementsAreArray(ok_params[i].expected_output));
   }
 }
+
+TEST_F(OpConverterTest, ConvertPool3D) {
+  // Get nodedef for MaxPool3D and AvgPool3D layers.
+  auto get_pool3d_nodedef = [](std::vector<int> ksize = {1, 1, 1, 1, 1},
+                               std::vector<int> strides = {1, 1, 1, 1, 1},
+                               string padding = "SAME",
+                               string data_format = "NCDHW",
+                               const bool is_max_pooling = true) -> NodeDef {
+    Scope s = Scope::NewRootScope();
+    auto input = ops::Placeholder(s.WithOpName("input"), DT_FLOAT);
+
+    if (is_max_pooling) {
+      ops::MaxPool3D::Attrs attrs =
+          ops::MaxPool3D::Attrs().DataFormat(data_format);
+      auto pool3d = ops::MaxPool3D(s.WithOpName("my_maxpool3d"), input, ksize,
+                                   strides, padding, attrs);
+      return pool3d.operation.node()->def();
+    } else {
+      ops::AvgPool3D::Attrs attrs =
+          ops::AvgPool3D::Attrs().DataFormat(data_format);
+      auto pool3d = ops::AvgPool3D(s.WithOpName("my_avgpool3d"), input, ksize,
+                                   strides, padding, attrs);
+      return pool3d.operation.node()->def();
+    }
+  };
+
+  {
+    // Input is weights, should fail.
+    Reset();
+    NodeDef node_def = get_pool3d_nodedef();
+
+    AddTestWeights<float>("input", {1, 2, 3}, {1, 2, 3, 4, 5, 6});
+    RunValidationAndConversion(
+        node_def, error::UNIMPLEMENTED,
+        "The input \"input\" for MaxPool3D must be a tensor, at my_maxpool3d");
+  }
+
+  struct TestParams {
+    std::vector<int> input_dims;
+    std::vector<float> input;
+    std::vector<int> ksize;
+    std::vector<int> strides;
+    string padding;
+    string data_format;
+    bool is_max_pooling;
+    std::vector<int> expected_output_dims;
+    std::vector<float> expected_output;
+  };
+
+  // Start here
+  const int kPool3DOKCases = 10;
+  TestParams ok_params[kPool3DOKCases] = {
+      // Basic - just 1x1 max pooling - input = output
+      TestParams{
+          /*input_dims=*/{1, 3, 3, 3},
+          /*input=*/{1, 2,  15,  3, 6,  -3, 22, 1, 88, 56, 36, 1,  1, 105,
```

Comment:
nit: this input is repeated many times, can we make it a `const vector<float>` and reuse it for all occurrences?

Reply:
Addressed in https://github.com/tensorflow/tensorflow/pull/32843/commits/ab48f4961cc0e3fb31085b6a888dd38cbc9e80fa .